### PR TITLE
chore(pie-monorepo): DSW-1540 extract create and publish snapshot step into its own JS file

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Describe your changes (can list changeset entries if preferable)
- Extract the script into its own file
- Write tests for the script (!)
- Update the comment to include guidance to update all packages

There's a reasonably high chance of this breaking when we extract it, but I followed these docs for the github-script action: https://github.com/actions/github-script/tree/v6/?tab=readme-ov-file#run-a-separate-file

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: .changeset/ninety-panthers-cough.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@justeattakeaway/pie-button": patch
+---
+
+[Changed] - Test change please ignore
```
Comment (ashleynolan) [inline .changeset/ninety-panthers-cough.md:5]: Guessing you'll be removing this before merging 😂

**Review by ashleynolan — COMMENTED** [review]
Looks good - just need to remove the sample changeset

File: .changeset/ninety-panthers-cough.md (hunk 1 above)
Comment (xander-marjoram) [reply .changeset/ninety-panthers-cough.md:5]: Thanks for the reminder 😄

**Review by jamieomaguire — APPROVED** [review]

**Review by kevinrodrigues — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]

**Comment by jamieomaguire** [comment]
@xander-marjoram this work is fantastic, I am temporarily requesting a revert just because there is some issue with snapit not being able to locate the new JS file when people attempt to run it. 

I want us to keep this change, we might just need to do a bit more testing in the new year (happy to help with this).

@ashleynolan please revert when ready 👍 

Screenshot 2023-12-22 at 14 27 57

**Comment by ashleynolan** [comment]
For reference – [this is the revert](https://github.com/justeattakeaway/pie/pull/1129). @xander-marjoram feel free to create a new PR with you branch when you're back in the new year so we can get it back in.

**Pull request merged**